internal/runtime/client/client.go

```diff
@@ -266,13 +264,15 @@ func aggregateSuccessfulResponses(aggregatedResponse runtimehooksv1.ResponseObje
 	}
 	// Note: As all responses have the same type we can assume now that
 	// they all implement the RetryResponseObject interface.
-
+	messages := []string{}
 	for _, resp := range responses {
 		aggregatedRetryResponse.SetRetryAfterSeconds(lowestNonZeroRetryAfterSeconds(
 			aggregatedRetryResponse.GetRetryAfterSeconds(),
 			resp.(runtimehooksv1.RetryResponseObject).GetRetryAfterSeconds(),
 		))
+		messages = append(messages, resp.GetMessage())
 	}
+	aggregatedResponse.SetMessage(strings.Join(messages, ", "))
```

Comment:
I know we want back and forth on this but if the extension's errors do not surface the UX degrades a lot.
So I'm proposing to surface errors, document guidelines for error messages, and take time to implement a protection mechanism for flickering conditions